### DISCUSSION TITLE
MySQL varchar => text via prisma migrate

### DISCUSSION THREAD

**Discussion by jkottapa** [prisma/prisma#2138] [Q&A] [answered]
I want to save some long text in a column of a table. However, I get the following error:
``` prisma
The provided value for the column is too long for the column's type. Column: location

model Order {
  id  Int @id @default(autoincrement())
  stId  String @unique
  subtotalPrice String @default("0.00")
  totalPrice  String @default("0.00")
  totalTax  String @default("0.00")
  location String

  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
}
```

Do I have to change the column type manually or is there a way to do that using `prisma migrate`

**Comment by tomhoule** [comment, marked as answer]
Migrate doesn't allow you to do that at the moment, so if you want a text column, you have to migrate it directly with SQL. We are very conscious of the problem and we will find a solution as we work on migrate again in the near future :)

**Reply by jurtjee1997** [reply]
any update?

**Comment by jkottapa** [comment]
Is there a way to write a custom migration to make this happen now?

**Reply by tomhoule** [reply]
At the moment no, not with prisma migrate, you would have to migrate it with SQL (manually or any SQL-based migration tool).

**Comment by jkottapa** [comment]
@tomhoule Tried your solution, but its resulting in another issue
```sh
⚠️  There will be data loss:

  • You are about to alter the column `location` on the `Order` table, which still contains 1 non-null values. The data in that column will be lost.
  • Made the column `location` on table `Order` required, but there are existing NULL values.
  • You are about to alter the column `location` on the `Order` table, which still contains 1 non-null values. The data in that column will be lost.
  • Made the column `location` on table `Order` required, but there are existing NULL values.
...
Failure during a migration command: Connector error. (error: Error querying the database: Error querying the database: Server error: `ERROR 42000 (1064): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'CURRENT_TIMESTAMP,
DROP COLUMN `location`,
ADD COLUMN `location` varchar(191) NO' at line 1')
```

To add more context. I changed the Column manually on MySQL but a future migration results in these messages and error.

**Reply by tomhoule** [reply]
The CURRENT_TIMESTAMP issue seems like a separate bug. Can you open an issue so we can track this?

Was `location` a TEXT column in your example? If so that's another bug (it should not migrate a text column to varchar), and another issue can be opened :) thanks for reporting this

**Reply by jkottapa** [reply]
Here is the first issue: https://github.com/prisma/prisma/issues/2286

I am not sure what its trying to do with the `location`, I've moved on to having a different representation anyway and dropped the column.

Not gonna lie, its been a bit frustrating to work with migrate (don't get me wrong, it has great potential). I will probably fallback on raw SQL for the time being.

**Reply by tomhoule** [reply]
Yes I can't recommend relying on migrate right now. If you keep using it in the near future, expect more breaking changes. We're also here for support to fix what doesn't work, but it is in a state of flow at the moment as we define exactly what we want the tool to be.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
